modules/intensity_transform/test/test_intensity_transform.cpp

```diff
@@ -0,0 +1,199 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+TEST(intensity_transform_logTransform, accuracy)
+{
+    cv::Mat const image = (cv::Mat_<uchar>(10, 10) <<
+                            51, 211, 212,  38,  48,  25, 189,  16,  64, 197,
+                           104, 137,  60,  10,  78, 234, 186, 149,  37, 236,
+                           128,  80,   6,  53,   7,  65, 233,  15, 216,  42,
+                           108, 132, 136, 194, 117, 128, 214,  46, 220, 119,
+                           101, 126, 148,  22,  86, 206,  91, 125, 234,  24,
+                           162, 136,  46, 247, 245,  81, 157, 126,  73, 173,
+                           120, 230, 117, 111, 145, 168, 169, 187,  23, 109,
+                             0, 184,  23,  43, 108, 201,  13, 170, 249, 228,
+                           107,  59,  73, 254, 116, 156, 209, 155, 149,  95,
+                            24, 245, 136, 107, 192, 114,  69,  80, 199,   8);
+
+    cv::Mat res;
+    cv::intensity_transform::logTransform(image, res);
+
+    cv::Mat const expectedRes = (cv::Mat_<uchar>(10, 10) <<
+                                 182, 247, 247, 169, 179, 150, 241, 130, 192, 243,
+                                 214, 227, 189, 110, 201, 251, 241, 231, 167, 252,
+                                 224, 202,  90, 184,  96, 193, 251, 128, 248, 173,
+                                 216, 225, 226, 243, 220, 224, 247, 177, 248, 220,
+                                 213, 223, 230, 144, 206, 245, 208, 223, 251, 148,
+                                 234, 226, 177, 254, 253, 203, 233, 223, 198, 237,
+                                 221, 250, 220, 217, 229, 236, 236, 241, 146, 216,
+                                   0, 240, 146, 174, 216, 244, 121, 237, 254, 250,
+                                 215, 188, 198, 255, 219, 233, 246, 232, 231, 210,
+                                 148, 253, 226, 215, 242, 218, 196, 202, 244, 101);
+
+    for(int i = 0; i < res.cols; ++i)
+    {
+        for(int j = 0; j < res.rows; ++j)
+        {
+            EXPECT_EQ((int)res.at<uint8_t>(i,j), (int)expectedRes.at<uint8_t>(i,j));
```

Comment:
loops =>
```
EXPECT_EQ(0, cvtest::norm(res, expectedRes, NORM_INF));
```

Please note that this check is very strong (and should be avoided due non bit-exact floating-point implementation).
It is better to relax conditions:
```
EXPECT_LE(cvtest::norm(res, expectedRes, NORM_INF), 1);
EXPECT_LE(cvtest::norm(res, expectedRes, NORM_L1), 10);
```